### ISSUE TITLE
Alternative way to retrieve the `recommendedNonce`

### ISSUE THREAD

**Issue by fmrsabino** [safe-global/safe-client-gateway#681]
Currently the `recommendedNonce` is retrieved by calling `CGW POST chains/:chainId/safes/:address/multisig-transactions/estimations`.

This route calls the following endpoints on the Safe Transaction Service:
1. `TX POST /api/v1/safes/:safeAddress/multisig-transactions/estimations/`
2. `TX GET /api/v1/safes/:safeAddress`
3. `TX GET /api/v1/safes/:safeAddress/multisig-transactions/` – with `ordering=-nonce&trusted=true&limit=1`

However the parameters returned by call 1 (`safeTxGas`) are not used by the frontend clients for 1.3.0 safes. This means that starting from Safes with version `1.3.0`, the recommended nonce is computed using the current Safe Nonce (from call 2) and the most recent nonce from a trusted Multisig contract (call 3).

## Proposal

Since we should keep compatibility with Safes with version lower than 1.3.0, `CGW POST chains/:chainId/safes/:address/multisig-transactions/estimations` should remain with the same functionality.

Given that for Safes with version greater than 1.3.0, the call to `TX POST /api/v1/safes/:safeAddress/multisig-transactions/estimations/` should be avoided, we should introduce a new endpoint or add the `recommendedNonce` to an existing endpoint.

One idea is to introduce `recommendedNonce` to `CGW GET chains/:chainId/safes/:safeAddress` however this is also one of the most requested endpoints and we might be computing data which is not required by the frontend when this endpoint is called.

**Comment by katspaugh** [comment]
Based on the discussion in the infra sync:

* It's important to migrate to an alternative recommended nonce retrieval soon because the estimation endpoint for this right now which incures a significant overhead/cost (as per @Uxio0)
* the current client-side usage is that we fetch the recommended nonce as a separate request when starting a tx flow (we show the nonce right away), so it would be good to keep it as a separate endpoint
* hovewer, we also poll this endpoint when the queue tag updates, so there's definitely a relation to the safe info/queue – so adding a new field there would also work

**Comment by fmrsabino** [comment]
From a domain perspective I believe that adding the `recommendedNonce` to the Safe Info endpoint would further convolute the endpoint and what is returned.

The reason is that `recommendedNonce` is mostly for transaction execution purposes and Safe Info can be seen as an entity void of any transaction execution related data (this is already not true due to the tags which are present).

So given the options described I'm leaning towards more a separate endpoint: one that would return the `recommendedNonce` for a specific Safe (regardless of the contents of Safe Info).

> * the current client-side usage is that we fetch the recommended nonce as a separate request when starting a tx flow (we show the nonce right away), so it would be good to keep it as a separate endpoint

So in conclusion I would follow the suggestion done here by @katspaugh ☝️ 

@Uxio0 @moisses89  This new endpoint would call `GET /api/v1/safes/<safe-address>/multisig-transactions/` of the transaction service with the following parameters: 

```
ordering: '-nonce'
trusted: true
limit: 1
```

This will be cached as with the other endpoints.

@safe-global/safe-client-gateway-maintainers @katspaugh what do you think about this and adding the following endpoint to the Safe Client Gateway:

```
GET /v1/chains/<chainId>/safes/<safeAddress>/recommended-nonce

{
  "nonce": <number>
}
```

**Comment by Uxio0** [comment]
We have deployed today a fix so estimation endpoint will return always `0` in L2 networks so we give you more time to implement this change. A new endpoint to get the nonce sounds good to me
